commonui/src/main/java/com/hyperwallet/android/ui/common/insight/HyperwalletInsight.java

```diff
@@ -18,34 +18,202 @@
+            @NonNull final String pageGroup, @NonNull final String link, @NonNull final Map<String, String> params) {
+
+        if (Insight.getInsightTracker().isInitialized()) {
+            Insight.getInsightTracker().trackImpression(context, pageName, pageGroup, params);
+
+        } else {
+            mExecutor.execute(new Runnable() {
+                @Override
+                public void run() {
+                    Hyperwallet.getDefault().getConfiguration(new HyperwalletListener<Configuration>() {
+                        @Override
+                        public void onSuccess(@Nullable Configuration result) {
+                            if (result != null) {
+                                onGetConfigurationSuccess(context, result);
+                                Insight.getInsightTracker().trackImpression(context, pageName, pageGroup, params);
+                            }
+                        }
+
+                        @Override
+                        public void onFailure(HyperwalletException exception) {
+                            // TODO unsure what to log here
+                            final String logMessage = MessageFormat.format(
+                                    "Unable to find Configuration using default token provider: {0}", exception);
+                        }
+
+                        @Override
+                        public Handler getHandler() {
+                            return null;
+                        }
+                    });
+                }
+            });
+        }
+    }
+
+    /**
+     * Used to track an error received by the user. This can be used for, and not limited to: system errors, caught
+     * exceptions, or validation input errors.
+     *
+     * @param context   Context where the tracking is happening
+     * @param pageName  an arbituary pageName the user is currently viewing as defined by the app
+     * @param pageGroup an arbituary pageGroup the user is currently viewing as defined by the app
+     * @param errorInfo additional error information that can be added to make the tracking event more useful
+     */
+    public void trackError(@NonNull final Context context, @NonNull final String pageName,
+            @NonNull final String pageGroup, @NonNull final ErrorInfo errorInfo) {
+
+        if (Insight.getInsightTracker().isInitialized()) {
+            Insight.getInsightTracker().trackError(context, pageName, pageGroup, errorInfo);
+
+        } else {
+            mExecutor.execute(new Runnable() {
+                @Override
+                public void run() {
+                    Hyperwallet.getDefault().getConfiguration(new HyperwalletListener<Configuration>() {
+                        @Override
+                        public void onSuccess(@Nullable Configuration result) {
+                            if (result != null) {
+                                onGetConfigurationSuccess(context, result);
```

Comment:
we can directly call `HyperwalletInsight.getInstance().initializeInsight(context, configuration)`